presto-main/src/main/java/com/facebook/presto/operator/GenericPageProcessor.java

```diff
@@ -46,6 +49,7 @@
 
     private final Block[] inputDictionaries;
     private final Block[] outputDictionaries;
+    private final ImmutableList<Type> types;
```

Comment:
In the old code `types` were passed to the `process` method.  In the new code, types are expected to have been stored in a field of the class.  So this is not an "optional" change and is instead part of this commit.